var/spack/repos/builtin/packages/swig/package.py

```diff
@@ -70,3 +70,45 @@ def create_symlink(self):
         @when(_version)
         def autoreconf(self, spec, prefix):
             which('sh')('./autogen.sh')
+
+    @property
+    def _installed_exe(self):
+        return join_path(self.prefix, 'bin', 'swig')
+
+    def _test_version(self):
+        version = str(self.version)
+        if version.endswith('-fortran'):
+            version = version.replace('-', r'\+')
+        elif version in ('fortran', 'master'):
+            version = r'.*'
+
+        self.run_test(self._installed_exe,
+                      '-version',
+                      expected=[r'SWIG Version {0}'.format(version)],
+                      purpose="test: version")
+
+    def _test_swiglib(self):
+        # Get SWIG's alleged path to library files
+        swig = Executable(self._installed_exe)
+        swiglib = swig('-swiglib', output=str).strip()
+
+        # Check that the lib dir exists
+        if not os.path.isdir(swiglib):
+            msg = "SWIG library does not exist at '{0}'".format(swiglib)
+            self.test_failures.append([None, msg])
+
+        # Check for existence of other critical SWIG library files
+        swigfile = join_path(swiglib, 'swig.swg')
+        if not os.path.exists(swigfile):
+            msg = "SWIG runtime does not exist at '{0}'".format(swigfile)
+            self.test_failures.append([None, msg])
+        if 'fortran' in str(self.version):
+            swigfile = join_path(swiglib, 'fortran', 'fortran.swg')
+            if not os.path.exists(swigfile):
+                msg = "SWIG+Fortran runtime does not exist at '{0}'".format(
+                    swigfile)
+                self.test_failures.append([None, msg])
```

Comment:
The concern I have with these checks is they are not advertised as being performed in the test log file.  I realize this relates to a lack of infrastructure, but only the version checks are being reported by `spack test results -l`.

Reply:
OK, is it sufficient to add a `print` statement at the end if they pass? Or is there a better way to test and add pass/fail messages?

Reply:
Well...  There is a temporary hack. It generates all the expected test (part) outputs BUT people aren't going to like it *and* it generates a warning.  Specifically, you *could* use `python` as follows until a better solution comes around:

```
        self.run_test('python',
                      ['-c',
                       '"import os; assert os.path.exists({0})"'.format(swiglib)],
                      expected=[],
                      purpose="test: ensuring swig lib dir exists")
```

Reply:
😱 😂

Reply:
Darn!  Nvm.  It will pass whether the path exists or not. :cry:  (The warning should've been a give-away.)

Reply:
Another hack, off the top of my head, is to write a script that takes the path as an argument.  BUT I haven't tried that yet.  

Perhaps this is a rabbit hole not worth going down.  I've made a note to revisit this package when implementing an assert-like mechanism.

Reply:
I'll approve and merge as is unless you want to try something else (e.g., a `print`).

Reply:
Great! I think that as long as the smoke test *catches* any broken installation, it's doing 95% of its job. I think that the current state does that well enough; printing progress for a successful test is usually icing on the cake.